Userland/Libraries/LibJS/Runtime/Intl/NumberFormat.cpp

```diff
@@ -455,6 +471,664 @@ int currency_digits(StringView currency)
+    }
+
+    // 11. If isNegative, then
+    if (is_negative) {
+        // a. Let x be -x.
+        number *= -1;
+    }
+
+    // 12. Return the Record { [[RoundedNumber]]: x, [[FormattedString]]: string }.
+    return { move(string), number };
+}
+
+// 15.1.6 PartitionNumberPattern ( numberFormat, x ), https://tc39.es/ecma402/#sec-partitionnumberpattern
+Vector<PatternPartition> partition_number_pattern(NumberFormat& number_format, double number)
+{
+    // 1. Let exponent be 0.
+    int exponent = 0;
+
+    String formatted_string;
+
+    // 2. If x is NaN, then
+    if (Value(number).is_nan()) {
+        // a. Let n be an implementation- and locale-dependent (ILD) String value indicating the NaN value.
+        formatted_string = Unicode::get_number_system_symbol(number_format.data_locale(), number_format.numbering_system(), "nan"sv).value_or("NaN"sv);
+    }
+    // 3. Else if x is a non-finite Number, then
+    else if (!Value(number).is_finite_number()) {
+        // a. Let n be an ILD String value indicating infinity.
+        formatted_string = Unicode::get_number_system_symbol(number_format.data_locale(), number_format.numbering_system(), "infinity"sv).value_or("infinity"sv);
+    }
+    // 4. Else,
+    else {
+        // a. If numberFormat.[[Style]] is "percent", let x be 100 × x.
+        if (number_format.style() == NumberFormat::Style::Percent)
+            number = number * 100;
+
+        // b. Let exponent be ComputeExponent(numberFormat, x).
+        exponent = compute_exponent(number_format, number);
+
+        // c. Let x be x × 10^(-exponent).
+        number *= pow(10, -exponent);
+
+        // d. Let formatNumberResult be FormatNumericToString(numberFormat, x).
+        auto format_number_result = format_numeric_to_string(number_format, number);
+
+        // e. Let n be formatNumberResult.[[FormattedString]].
+        formatted_string = move(format_number_result.formatted_string);
+
+        // f. Let x be formatNumberResult.[[RoundedNumber]].
+        number = format_number_result.rounded_number;
+    }
+
+    // 5. Let pattern be GetNumberFormatPattern(numberFormat, x).
+    auto pattern = get_number_format_pattern(number_format, number);
+    if (!pattern.has_value())
+        return {};
+
+    // 6. Let result be a new empty List.
+    Vector<PatternPartition> result;
+
+    // 7. Let patternParts be PartitionPattern(pattern).
+    auto pattern_parts = partition_pattern(*pattern);
+
+    // 8. For each Record { [[Type]], [[Value]] } patternPart of patternParts, do
+    for (auto& pattern_part : pattern_parts) {
+        // a. Let p be patternPart.[[Type]].
+        auto part = pattern_part.type;
+
+        // b. If p is "literal", then
+        if (part == "literal"sv) {
+            // i. Append a new Record { [[Type]]: "literal", [[Value]]: patternPart.[[Value]] } as the last element of result.
+            result.append({ part, move(pattern_part.value) });
+        }
+
```

Comment:
Please omit newlines between if/else branches

Reply:
TBH this section is so long that it just looked way too squished together for me, made my old eyes feel pretty strained, so I did these newlines on purpose.

Reply:
ok, fair enough. Don't tell @alimpfard though. :^)

Reply:
I compromised on a few of these that weren't so bad squished together :)